src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/MapCodecs.java

```diff
@@ -77,10 +77,13 @@ private static Map<Object, Object> toCqlMap(
           (entry.getKey() instanceof JsonLiteral<?> jsonLiteralKey)
               ? jsonLiteralKey.value()
               : entry.getKey();
+      // If the raw map key or value is JsonLiteral<>(null, JsonType.NULL);
       Object element = entry.getValue().value();
-      if (element == null) {
-        result.put(key, null);
-        continue;
+      if (key == null || element == null) {
+        throw new ToCQLCodecException(
+            null,
+            DataTypes.mapOf(keyType, elementType),
+            "null keys/values are not allowed in map column");
```

Comment:
nit: avoid using "/" like this in error messages it is often not clear. 

"null values not supported for the key or value of a map column"